Source/WebInspectorUI/UserInterface/Views/SourcesNavigationSidebarPanel.js

```diff
@@ -1404,7 +1404,14 @@ WI.SourcesNavigationSidebarPanel = class SourcesNavigationSidebarPanel extends W
 
     _addBreakpoint(breakpoint)
     {
-        if (this._breakpointsTreeOutline.findTreeElement(breakpoint))
+        let existingTreeElement = this._breakpointsTreeOutline.findTreeElement(breakpoint);
+        let newSourceCode = breakpoint.sourceCodeLocation.displaySourceCode;
+        let shouldSkipAdding = !!existingTreeElement;
+        if (newSourceCode && breakpoint instanceof WI.JavaScriptBreakpoint && existingTreeElement?.parent.resource !== newSourceCode) {
+            // Normally, we can skip adding a new tree element if there already exists one representing the same breakpoint. But not if that tree element's parent has a different (older) version of source code, which can happen due to a page-reload with the inspector open. In that case, that parent element will be removed by the next `NavigationSidebarPanel.pruneStaleResourceTreeElements()`, so we must not pretend that the found element will continue existing and must add a new element instead.
```

Comment:
this seems more like a bug with `pruneStaleResourceTreeElements` in that it's not happening soon enough.  i wouldn't be surprised if there's other bugs like this elsewhere that would also benefit from a fix

maybe we need to trigger that in `WI._frameWasAdded` and/or `WI._mainFrameDidChange` instead of or in addition to in `WI._mainResourceDidChange`?